frontend/src/metabase/components/SingleSelectListField/SingleSelectListField.tsx

```diff
@@ -63,15 +68,21 @@ const SingleSelectListField = ({
   }, [augmentedOptions.length]);
 
   const [filter, setFilter] = useState("");
-  const waitTime = useContext(waitTimeContext);
-  const debouncedFilter = useDebouncedValue(filter, waitTime);
+  const debouncedFilter = useDebouncedValue(filter, DEBOUNCE_FILTER_TIME);
+
+  const isFilterInValues = value[0] === filter;
 
   const filteredOptions = useMemo(() => {
     const formattedFilter = debouncedFilter.trim().toLowerCase();
     if (formattedFilter.length === 0) {
       return sortedOptions;
     }
 
+    // Allow picking of different values in the list
+    if (isFilterInValues) {
+      return augmentedOptions;
+    }
```

Comment:
I think this is still incorrect.

"booster" should not be displayed [here](https://github.com/user-attachments/assets/f1744d48-35b3-4a21-99f0-4c75feafb7dc).

----


https://github.com/user-attachments/assets/44025773-5b1a-44e6-92da-f12c1ea6bb3c

Reply:
I agree that that's confusing but it's not really a regression since that just how the search for these filters work (even before this PR).

There's something funky going on with the debouncing and I think we should fix that but I think that's out of the scope of this PR. 

https://github.com/user-attachments/assets/71581312-def7-44d7-9304-d15e926d1da5